common/djangoapps/util/request.py

```diff
@@ -7,8 +7,8 @@
 from opaque_keys import InvalidKeyError
 from opaque_keys.edx.locations import SlashSeparatedCourseKey
 
-
-COURSE_REGEX = re.compile(r'^.*?/courses/{}'.format(settings.COURSE_ID_PATTERN))
+# accommodates course api urls, excluding any course api routes that do not fall under v*/courses, such as v1/blocks.
+COURSE_REGEX = re.compile(r'^(.*?/courses/)+(?!v[0-9]+/[^/]+){}'.format(settings.COURSE_ID_PATTERN))
```

Comment:
But you already handle that case by specifying that you want to match `/courses/` not followed by `v[0-9]+`, right?